### PR TITLE
[Issue-2118] Update transferable balance

### PR DESCRIPTION
**Pull request by bluezdot**
### Related issue(s)
#2118 

### Is your feature request related to a problem(s)? Please describe.

- Update code for transferable balance formula

### Describe the solution you'd like

- Update some types of balance re-defined by Polkadot.

### Describe alternatives you've considered

- No

### Additional context

- No

### CODE REVIEW

File: packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts [hunk 1]
```diff
@@ -97,36 +97,47 @@ async function subscribeWithSystemAccountPallet (addresses: string[], chainInfo:
     }
 
     const items: BalanceItem[] = balances.map((balance: AccountInfo, index) => {
-      let total = balance.data?.free?.toBn() || new BN(0);
+      const free = balance.data?.free?.toBn() || new BN(0);
       const reserved = balance.data?.reserved?.toBn() || new BN(0);
       // @ts-ignore
       // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
-      const miscFrozen = balance.data?.miscFrozen?.toBn() || balance?.data?.frozen?.toBn() || new BN(0);
+      const locked = balance.data?.miscFrozen?.toBn() || balance?.data?.frozen?.toBn() || new BN(0);
```
Comment (NamPhamc99) [inline packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts:104]: Need to check when it's `balance.data.frozen` or `balance.data.miscFrozen`

File: packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts [hunk 2]
```diff
@@ -97,36 +97,47 @@ async function subscribeWithSystemAccountPallet (addresses: string[], chainInfo:
...
 
-      let locked = reserved.add(miscFrozen);
-
-      total = total.add(reserved);
-
+      let frozen = reserved.add(locked);
+      let total = free.add(reserved);
       const pooledStakingBalance = pooledStakingBalances[index] || BN_ZERO;
 
+      console.log("chain: ", chainInfo.slug);
```
Comment (NamPhamc99) [inline packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts:111]: Delete console.log

File: packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts [hunk 3]
```diff
@@ -97,36 +97,47 @@ async function subscribeWithSystemAccountPallet (addresses: string[], chainInfo:
...
+
       if (pooledStakingBalance.gt(BN_ZERO)) {
         total = total.add(pooledStakingBalance);
-        locked = locked.add(pooledStakingBalance);
+        frozen = locked.add(pooledStakingBalance);
       }
 
-      const free = total.sub(locked);
-
+      console.log("new frozen: ", frozen.toString());
```
Comment (NamPhamc99) [inline packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts:126]: Remove console.log

**Review by NamPhamc99 — CHANGES_REQUESTED** [review]
All good, only need a few adjustments

File: packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts (hunk 1 above)
Comment (bluezdot) [reply packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts:104]: Currently, I see only some chains: edgeware, joystream, terno have balance.data.miscFrozen. The common point of these chains is belong to "Live Networks" on Polkadot(.js) UI.

Comment (NamPhamc99) [reply packages/extension-base/src/services/balance-service/helpers/subscribe/substrate/index.ts:104]: Mark TODO on this line for later reviews

**Comment by saltict** [comment]
🚀 Deployed on https://pr-2303--sw-web-runner.netlify.app

**Comment by nampc1** [comment]
This issue will be resolved by https://github.com/Koniverse/SubWallet-Extension/pull/3042

**Pull request closed without merging**